aks.tf

```diff
@@ -0,0 +1,89 @@
+resource "azurerm_kubernetes_cluster" "aks" {
+  lifecycle {
+    ignore_changes = [
+      default_node_pool[0].node_count
+    ]
+  }
+
+  name                       = local.cluster_name
+  enable_pod_security_policy = false
+  location                   = local.resource_group.location
+  resource_group_name        = local.resource_group.name
+  dns_prefix                 = local.cluster_name
+  private_cluster_enabled    = false
+
+  api_server_authorized_ip_ranges = var.api_server_authorized_ip_ranges
+
+  default_node_pool {
+    enable_node_public_ip = var.node_pools.platform.enable_node_public_ip
+    name                  = "platform"
+    node_count            = var.node_pools.platform.min_count
+    node_labels           = var.node_pools.platform.node_labels
+    vm_size               = var.node_pools.platform.vm_size
+    availability_zones    = var.node_pools.platform.zones
+    os_disk_size_gb       = var.node_pools.platform.os_disk_size_gb
+    node_taints           = var.node_pools.platform.node_taints
+    enable_auto_scaling   = var.node_pools.platform.enable_auto_scaling
+    min_count             = var.node_pools.platform.min_count
+    max_count             = var.node_pools.platform.max_count
+    max_pods              = var.node_pools.platform.max_pods
+    tags                  = {}
+  }
+
+  identity {
+    type = "SystemAssigned"
+  }
+
+  addon_profile {
+    kube_dashboard {
+      enabled = false
+    }
+
+    oms_agent {
+      enabled                    = true
+      log_analytics_workspace_id = azurerm_log_analytics_workspace.logs.id
+    }
+  }
+
+  network_profile {
+    load_balancer_sku  = "Standard"
+    network_plugin     = "azure"
+    network_policy     = "calico"
+    dns_service_ip     = "100.97.0.10"
+    docker_bridge_cidr = "172.17.0.1/16"
+    service_cidr       = "100.97.0.0/16"
+  }
+
+  tags = var.tags
```

Comment:
shouldn't we have a set of default tags that we apply regardless of whatever is passed in?

Reply:
What kinds of tags are you thinking? `provisioned-by: domino` stuff like that?

Reply:
or even the name of the deployment or any details which are applied to all the resources since we don't own everything in the RG or network.

Reply:
Added a default tag for now. We should definitely pass in `var.tags` for stuff we control.